include/range/v3/at.hpp

```diff
@@ -30,7 +31,20 @@ namespace ranges
         {
             /// \return `begin(rng)[n]`
             template<typename Rng,
-                CONCEPT_REQUIRES_(RandomAccessRange<Rng>())>
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() and
+                                  SizedRange<Rng>())>
+            RANGES_CXX14_CONSTEXPR
+            auto operator()(Rng &&rng, range_difference_type_t<Rng> n) const ->
+                decltype(begin(rng)[n])
+            {
+                RANGES_EXPECT(static_cast<std::size_t>(n) < ranges::size(rng));
+                return begin(rng)[n];
+            }
+
+            /// \return `begin(rng)[n]`
+            template<typename Rng,
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() and
+                                  not SizedRange<Rng>())>
```

Comment:
> That's because you didn't replace the call to `ranges::size` with a call to `ranges::distance`.

Which is necessary for correctness: `ranges::distance` is valid for all ranges, `ranges::size` is not. Since that `if` isn't `if constexpr`, the code needs to use `distance` instead of `size`.